src/pkg/services/m365/api/groups.go

```diff
@@ -135,10 +139,19 @@ func (c Groups) GetByID(
 		logger.CtxErr(ctx, err).Info("finding group by id, falling back to display name")
 	}
 
+	var filterTemplate string
+	// attempt to find by email address if the identifier looks like an email
+	if isEmail(identifier) {
+		filterTemplate = filterGroupByMailQueryTmpl
+	} else {
+		filterTemplate = filterGroupByDisplayNameQueryTmpl
+	}
```

Comment:
I think we'll want to check both cases instead of a conditional.  afaik, nothing prevents a user from naming a group with a pattern that matches emails.

Reply:
roger. done!